### PR TITLE
Add mean absolute percentage error

### PR DESCRIPTION
**Pull request by Shashi456**
Reference #127

### CODE REVIEW

File: Sources/TensorFlow/Loss.swift [hunk 1]
```diff
@@ -53,6 +53,19 @@ public func meanAbsoluteError<Scalar: TensorFlowFloatingPoint>(
...
+/// Computes the mean absolute percentage error between predictions and expectations.
+///
+/// - Parameters:
+///   - predicted: Predicted outputs from a neural network.
+///   - expected: Expected values, i.e. targets, that correspond to the correct output.
+@differentiable(wrt: predicted)
+public func meanAbsolutePercentageError<Scalar: TensorFlowFloatingPoint>(
+    predicted: Tensor<Scalar>, expected: Tensor<Scalar>
+) -> Tensor<Scalar> {
+    let diff = abs((expected - predicted) / abs(expected))
```
Comment (jon-tow) [inline Sources/TensorFlow/Loss.swift]: Just wondering. Could we avoid the call to abs() in the denominator?

Comment (Shashi456) [reply Sources/TensorFlow/Loss.swift]: @jon-tow I'm afraid not if you don't take the absolute value and you have a negative sign in one of your values then your percentages might no longer be uniform.

Comment (jon-tow) [reply Sources/TensorFlow/Loss.swift]: If you remove the abs() in the denominator of line 65, then the code is computationally equivalent to the equation you've commented above.

The code currently stands as: 
     abs(num / abs(denom))
The abs(denom) seems redundant. How does this break some uniformity?

Comment (Shashi456) [reply Sources/TensorFlow/Loss.swift]: A variety of loss functions have to be clipped. I'm waiting for #198 to pass to update the loss functions, in that case an abs function is needed. I guess we might not need the abs in this PR. I'll remove it in a while.

Comment (jon-tow) [reply Sources/TensorFlow/Loss.swift]: Ah I see. Thanks for the quick response and explanation. 👍

Comment (Shashi456) [reply Sources/TensorFlow/Loss.swift]: Thanks for pointing it out.

File: Sources/TensorFlow/Loss.swift [hunk 2]
```diff
@@ -53,6 +53,19 @@ public func meanAbsoluteError<Scalar: TensorFlowFloatingPoint>(
     return abs(expected - predicted).mean()
 }
 
+/// Computes the mean absolute percentage error between predictions and expectations.
```
Comment (rxwei) [inline Sources/TensorFlow/Loss.swift]:
```suggestion
/// Returns the mean absolute percentage error between predictions and expectations.
```

File: Tests/TensorFlowTests/LossTests.swift [hunk 3]
```diff
@@ -49,6 +49,16 @@ final class LossTests: XCTestCase {
...
 
+    func testMeanAbsolutePercentageError() {
+        let predicted = Tensor<Float>([1, 2, 3, 4, 5, 6, 7, 8])
+        let expected = Tensor<Float>([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8])
+
+        let loss = meanAbsolutePercentageError(predicted: predicted, expected: expected)
+        let expectedLoss: Float = 900.0
+        assertElementsEqual(expected: Tensor(expectedLoss), actual: loss)
+    }
+
```
Comment (rxwei) [inline Tests/TensorFlowTests/LossTests.swift]: Remove extra line
```suggestion
```

**Comment by Shashi456** [comment]
@rxwei this is ready to go.

File: Sources/TensorFlow/Loss.swift [hunk 4]
```diff
@@ -53,6 +53,19 @@ public func meanAbsoluteError<Scalar: TensorFlowFloatingPoint>(
...
+///
+/// - Parameters:
+///   - predicted: Predicted outputs from a neural network.
+///   - expected: Expected values, i.e. targets, that correspond to the correct output.
+@differentiable(wrt: predicted)
+public func meanAbsolutePercentageError<Scalar: TensorFlowFloatingPoint>(
+    predicted: Tensor<Scalar>, expected: Tensor<Scalar>
+) -> Tensor<Scalar> {
+    let diff = abs((expected - predicted) / abs(expected))
+    return Tensor(100) * diff.mean()
```
Comment (rxwei) [inline Sources/TensorFlow/Loss.swift]:
```suggestion
    return 100 * diff.mean()
```

**Comment by Shashi456** [comment]
@rxwei this is ready to go as well.
